nbextensions/usability/freeze/main.js

```diff
@@ -0,0 +1,126 @@
+define([
+	'base/js/namespace',
+	'base/js/events',
+	'notebook/js/codecell'
+], function (
+	Jupyter,
+	events,
+	codecell
+){
+	'use strict';
+	
+	var CodeCell = codecell.CodeCell;
+	
+	function patch_CodeCell_execute () {
+		console.log('[Freeze] patching CodeCell.prototype.execute')	
+		CodeCell.prototype.old_execute = CodeCell.prototype.execute;
```

Comment:
when patching objects, it's best not to attach your old values back onto the object - it can cause problems if another extension tries to patch the same method, and overwrites the `old_execute` value. Instead, it's safer use a variable local to either the `patch_CodeCell_execute` method, or the extension definition method.